spec/controllers/families_controller_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe FamiliesController do
+  describe "GET /index" do
+    it "is successful" do
+      get :index
+
+      expect(response).to be_success
+    end
+
+    it "renders the :index view" do
+      get :index
+
+      expect(response).to render_template :index
+    end
+
+    it "returns all the families" do
+      create_list(:family, 2)
+
+      get :index
+
+      expect(assigns[:families].size).to eq 2
+    end
+  end
+  
+  describe "POST /families" do
+    it "creates a new family with correct attribute" do
+      family_params = attributes_for(:family, last_name: "Scaramella", capability: "2")
+
+      post :create, params: { family: family_params }
+
+      family = Family.last
+
+      expect(family.last_name).to eq "Scaramella"
+    end
+
+    it "increases the count of the families by one" do
+      family_params = attributes_for(:family, last_name: "Scaramella", capability: "2")
+
+      expect { post :create, params: { family: family_params } }.to change(Family, :count).by(1)
+    end
+  end
+end
```

Comment:
This is just an information comment:
I've used some shoulda matchers that I found for my routing specs:
it { is_expected.to route(:get, "/users").to(controller: :users, action: :index) }
I am interested to hear Elle's opinion on their use - it made for smaller specs!

Reply:
Here I was tryint to test if a new family is created, your line is testing instead familes#index?! Not sure if I get you rcomment right.. 🤔

Reply:
Sorry, my users#index spec is just an example of a different and shorter way I found to do the routing specs, I have put it in my most recent PR to get Elle's opinion. I wasn't suggesting you change anything, it was just to show it to you.

Reply:
Ahhh, sorry, of course I didn't understand! 😅 Once I will be able to test against the model I will try it, thanks!

Reply:
I would usually have:
1. feature specs for seeing all families, and creating a new family (happy path)
2. controller specs for the sad path - i.e. trying to create a new family without required attributes => testing to see that the flash message was set correctly, and the rendered template is `:new` rather than `:index`, and that the object has error messages on it

Reply:
I created the folder for features specks and tested for add a family and vie all the family. I left in the controller specs the sad path.

Thanks!